src/components/Wizard/NewEditImportPane.tsx

```diff
@@ -108,11 +122,68 @@ function NewForm({ onNewMessage }: NewFormProps) {
   );
 }
 
-function EditForm() {
+function EditForm({ onEditMessage, onDeleteMessage, messages }: EditFormProps) {
+  const formContext = useForm<EditFormData>({ mode: "onChange" });
+  const { register, handleSubmit } = formContext;
+
+  const onSubmit: SubmitHandler<EditFormData> = (data: EditFormData) => {
+    onEditMessage(data.id);
+  };
+
+  const onDelete = (id: string) => {
+    onDeleteMessage(id);
+  };
   return (
     <>
       <Card.Title>Edit an Existing Message</Card.Title>
-      <Card.Text>There are no messages saved.</Card.Text>
+
+      <Form onSubmit={handleSubmit(onSubmit)} className="edit-form">
+        <FormProvider {...formContext}>
+          <Form.Group controlId="message-select">
+            <FormRow
+              label="Select Message"
+              containerClassName="form-input-check"
+            >
+              {Object.keys(messages).length ? (
+                Object.entries(messages).map(([id]) => {
+                  return (
+                    <Row className="edit-entry" key={id}>
+                      <RegisteredFormCheck
+                        name="id"
+                        register={register}
+                        registerOptions={{ required: true }}
+                        type="radio"
+                        key={id}
+                        label={id}
+                        value={id}
+                        id={`message-select-${id}`}
+                      />
+                      <div className="delete-col">
+                        <Button
+                          key={`delete-${id}`}
+                          type="button"
+                          variant="danger"
+                          onClick={() => onDelete(id)}
+                        >
+                          <FontAwesomeIcon icon={faTrash} />
+                        </Button>
+                      </div>
+                    </Row>
+                  );
+                })
+              ) : (
+                <Card.Text>There are no messages saved.</Card.Text>
+              )}
+            </FormRow>
+          </Form.Group>
+        </FormProvider>
+
+        <div className="form-row form-buttons">
+          <Button type="submit" disabled={!Object.keys(messages).length}>
```

Comment:
If you re-import `isDirty` and `isValid` from formContext.fromState you can do:

`disabled={!isDirty || !isValid}`

Which wont let you click the button until you select a message.